localize-templates.py

```diff
@@ -0,0 +1,97 @@
+#! /usr/bin/env python3
+'''
+This script aims to replace unlocalised templates in localised pages with the localised templates.
+'''
+import os
+import copy
+from ws.client import API
+import ws.ArchWiki.lang as lang
+import mwparserfromhell as mw
+
+
+localised_lang_name: str = "简体中文" # not required, can omit val; simplified Chinese (zh-Hans)
+templates: list[str] = ["Translateme"]
+api_url = "https://wiki.archlinux.org/api.php"
+index_url = "https://wiki.arclinux.org/index.php"
+cookie_path = os.path.expanduser("~/.cache/ArchWiki.cookie")
+session = API.make_session(ssl_verify=True,
+                           cookie_file=cookie_path)
+
+
+def edit(api: API, timestamp: str, text: str, page, template: str, language: str):
+    code = mw.parse(text)
+    for curTemplate in code.filter_templates():
+        if curTemplate.name.matches(template):
+            curTemplate.name = f"{template} ({language})"
+    text = str(code)
+    api.edit(pageid=page['pageid'], title=page['title'], basetimestamp=timestamp, text=new, summary="localize templates", bot="")
+    print(f"Edited {page['title']}")
+
+
+def process(template: str, language: str):
+    pages = []
+    texts: list[str] = []
+    timestamps: list[str] = []
+    api = API(api_url, index_url, session)
+    kwargs = {
+        "action": "query",
+        "format": "json",
+        "maxlag": "100",
+        "prop": "revisions",
+        "continue": "",
+        "generator": "embeddedin",
+        "redirects": 1,
+        "utf8": 1,
+        "formatversion": "2",
+        "rvprop": "content|timestamp",
+        "rvslots": "main",
+        "geititle": f"Template:{template}",
+        "geifilterredir": "nonredirects",
+        "geilimit": "max"
+    }
+    # get page content
+    print(end="Calling API: query generator...", flush=True)
+    ret = api.call_api(**kwargs)
```

Comment:
This will not work if there are too many results and the response is shortened to satisfy the limit (`geilimit`) – in this case we need to do a [query continuation](https://www.mediawiki.org/wiki/API:Query#Example_4:_Continuing_queries). You have the `continue` parameter, but not making multiple requests. You can use [api.query_continue](http://lahwaacz.github.io/wiki-scripts/ws/ws.client.api.html#ws.client.api.API.query_continue) or even [api.generator](http://lahwaacz.github.io/wiki-scripts/ws/ws.client.api.html#ws.client.api.API.generator) – I think there are some examples for the latter...

Note that query-continuation will also make sure that all pages in the result of `api.query_continue` and `api.generator` will contain all requested properties (such as the content). So the weird `try: ... except` block below, where you were trying to solve this, should not be necessary.